src/unix/process.c

```diff
@@ -197,10 +200,18 @@ static int uv__process_init_stdio(uv_stdio_container_t* container, int fds[2]) {
 
   case UV_CREATE_PIPE:
     assert(container->data.stream != NULL);
+
+    for (j = 0; j < i; j++) {
+      if (options->stdio[j].flags & UV_CREATE_PIPE && options->stdio[j].data.stream == container->data.stream) {
+        pipes[i][1] = pipes[j][1];
```

Comment:
I think this could result in double-closing the same file descriptor around line 576? That's bad in a multi-threaded program because the file descriptor might have been repurposed by another thread in the time window between the two close() calls; i.e., it ends up closing a file descriptor that doesn't belong to us.

I also wonder if using a single fd is really what you want rather than a dup()-licated fd. E.g., if the child process sets the O_CLOEXEC or O_NONBLOCK flag on its stdout fd, then that also affects the stderr fd, and vice versa.

Reply:
The fork needs to make the `dup2` call and set the options on those—I don't think there's anything to be done at this point in the code. (fwiw, O_NONBLOCK is a property of the file description not the descriptor, so that's an unavoidable problem)

Reply:
> I think this could result in double-closing the same file descriptor around line 576?

:heavy_check_mark: Fixed.

> I also wonder if using a single fd is really what you want rather than a dup()-licated fd.

The child process does [dup()-licate the fd](https://github.com/libuv/libuv/blob/5cb8860b45b445ebb3e5961b0e06c720ae0d42f2/src/unix/process.c#L328). With this change there might be two or more dup()-licates of the same fd, but each should be independent, so stdout flags shouldn't affect stderr, etc.